golang/src/metric_registrar_examples/main.go

```diff
@@ -1,69 +1,82 @@
+var simpleCounter = promauto.NewCounter(prometheus.CounterOpts{
+	Name: "simple",
+	Help: "Increments when /simple handler is called",
+})
+
+var customCounter = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "custom",
+	Help: "Custom gauge used to scale app instances",
 })
 
 var requestDuration = promauto.NewHistogramVec(
-    prometheus.HistogramOpts{
-        Name: "request_duration_seconds",
-        Help: "A histogram of latencies for requests.",
-    },
-    []string{"code", "method", "handler"},
+	prometheus.HistogramOpts{
+		Name: "request_duration_seconds",
+		Help: "A histogram of latencies for requests.",
+	},
+	[]string{"code", "method", "handler"},
 )
 
 func main() {
-    router := mux.NewRouter()
-    staticPath := os.Getenv("GOPATH") + "/src/metric_registrar_examples/static"
-
-    router.Handle("/", http.FileServer(http.Dir(staticPath)))
-    router.Handle("/metrics", promhttp.Handler())
+	router := mux.NewRouter()
+	staticPath := "./static"
 
-    router.HandleFunc("/simple", instrument(simple, "simple")).Methods(http.MethodGet)
-    router.HandleFunc("/high_latency", instrument(highLatency, "high_latency")).Methods(http.MethodGet)
-    router.HandleFunc("/custom_metric", instrument(customMetric, "custom_metric"))
+	router.Handle("/", http.FileServer(http.Dir(staticPath)))
+	router.Handle("/metrics", promhttp.Handler())
 
-    port := os.Getenv("PORT")
-    if port == "" {
-        port = "8080"
-    }
+	router.HandleFunc("/simple", instrument(simple, "simple")).Methods(http.MethodGet)
+	router.HandleFunc("/high_latency", instrument(highLatency, "high_latency")).Methods(http.MethodGet)
+	router.HandleFunc("/custom_metric", instrument(customMetric, "custom_metric"))
 
-    log.Fatal(http.ListenAndServe(":"+port, router))
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+	}
+	log.Printf("Starting server on port http://localhost:%s\n", port)
+	log.Fatal(http.ListenAndServe(":"+port, router))
 }
 
 func instrument(handlerFunc http.HandlerFunc, name string) http.HandlerFunc {
-    handlerDuration, err := requestDuration.CurryWith(prometheus.Labels{
-        "handler": name,
-    })
-    if err != nil {
-        panic(err)
-    }
-
-    return promhttp.InstrumentHandlerDuration(handlerDuration, handlerFunc)
+	handlerDuration, err := requestDuration.CurryWith(prometheus.Labels{
+		"handler": name,
+	})
+
+	if err != nil {
+		panic(err)
+	}
+
+	return promhttp.InstrumentHandlerDuration(handlerDuration, handlerFunc)
 }
 
 func simple(w http.ResponseWriter, _ *http.Request) {
-    w.Write([]byte("{}"))
+	simpleCounter.Inc()
+	w.Write([]byte("{}"))
 }
 
 func highLatency(w http.ResponseWriter, _ *http.Request) {
-    time.Sleep(2 * time.Second)
-    w.Write([]byte("{}"))
+	time.Sleep(2 * time.Second)
+	w.Write([]byte("{}"))
 }
 
-func customMetric(w http.ResponseWriter, _ *http.Request) {
-    customCounter.Inc()
-    w.Write([]byte("{}"))
+func customMetric(w http.ResponseWriter, r *http.Request) {
```

Comment:
The inverted condition here kind of obfuscates the fact that decrementing is a default case. Also, decrementing as a default case feels a little weird to me. 

Normally we'd expect a counter's value to monotonically increase, is decrementing even something we want to encourage?

Reply:
On the Java app, you were decrementing the gauge but only incrementing the counter.

Reply:
Ahh I should change the name, I changed to the type to a gauge but still called it a counter.